lib/haml/compiler.rb

```diff
@@ -265,6 +265,14 @@ def push_script(text, opts = {})
 
       unless block_given?
         push_generated_script(no_format ? "(#{text}\n).to_s" : build_script_formatter("(#{text}\n)", opts))
+        # TODO this \n should be able to be frozen but the following tests fails
+        #
+        # bin/rails test test/helper_test.rb:596
+        # bin/rails test test/engine_test.rb:534
+        #
+        # Both appear to be testing outer whitespace nuking but at this point
```

Comment:
If anyone has insight into this issue and is willing to help me fix the code/test to be able to freeze this string that would be much appreciated! 😍

Reply:
Unfortunately I have no time to assist you, but I would write many `binding.pry`s to see how string that shouldn't be freezed is freezed.